core/MemoryAccessInfo.hpp

```diff
@@ -48,7 +61,9 @@ namespace olympia {
             mmu_access_state_(MMUState::NO_ACCESS),
 
             // Construct the State object here
-            cache_access_state_(CacheState::NO_ACCESS) {}
+            cache_access_state_(CacheState::NO_ACCESS),
+            src_(ArchUnit::NO_ACCESS),
+            dest_(ArchUnit::NO_ACCESS) {}
```

Comment:
Can you please explain the semantics of src_ and dest_ of MemoryAccessInfo?

Reply:
I added these members to MemoryAccessInfo for keeping track of the flow of transaction.
When the request arrives in L2Cache, I create a MemoryAccessInfo packet where I set the `src_ = dest_ = requestor`.

- If it hits in L2, the packet is returned to` dest_`.
- If it misses, the `dest_ = BIU`.
- When BIU returns, `src_ = BIU` and `dest_` becomes the original src where it needs to be returned to.

I am using the `src_` and `dest_` to determine where the packet is headed after the cacheLookup for HIT, MISS or RELOAD.

One major reason to have something like this is that L2Cache model supports requests from both _ICache_ and _DCache_. This aids in keeping track of the original requestor.